app/controllers/api/v2/moves_actions.rb

```diff
@@ -0,0 +1,148 @@
+module Api::V2
+  module MovesActions
+    def index_and_render
+      moves = Moves::Finder.new(filter_params, current_ability, params[:sort] || {}).call
+
+      paginate moves,
+               each_serializer: ::V2::MoveSerializer,
+               include: included_relationships,
+               fields: ::V2::MoveSerializer::INCLUDED_FIELDS
+    end
+
+    def show_and_render
+      render_move(move, :ok)
+    end
+
+    def create_and_render
+      move = Move.new(move_attributes)
+      authorize!(:create, move)
+      move.save!
+
+      Notifier.prepare_notifications(topic: move, action_name: 'create')
+
+      render_move(move, :created)
+    end
+
+    def update_and_render
+      raise ActiveRecord::ReadOnlyRecord, "Can't change moves coming from Nomis" if move.from_nomis?
+
+      @move.assign_attributes(move_attributes)
+      action_name = @move.status_changed? ? 'update_status' : 'update'
+      @move.save!
+      @move.allocation&.refresh_status_and_moves_count!
```

Comment:
My worries here are that if we start putting logic in the controller, we give permission to the developers who will come after us, to put a couple of lines more, and some more, and so on 😱 

That's why I think an `updater` can actually be good to have.
Also, it can speed up controller tests.

Reply:
How do you feel if we add a ticket to do that separately?

Reply:
I'm thinking:

1. Get this out and ready for integration
2. Update swagger documentation
3. Update supplier release documentation
4. Add a v2 updater service, unit test that and reduce the burden on the v2 update specs

Reply:
yes, that's fine, it make sense we are a bit in a rush.
Let's create a ticket to clean up this little bit.